lib/metadataTypes/MetadataType.js

```diff
@@ -468,31 +491,54 @@ class MetadataType {
+     * @param {boolean} [handleOutside] if the API reponse is irregular this allows you to handle it outside of this generic method
      * @returns {Promise} Promise
      */
-    static async updateSOAP(metadataEntry) {
+    static async updateSOAP(metadataEntry, overrideType, handleOutside) {
+        let response;
         try {
-            const res = await new Promise((resolve, reject) => {
-                this.removeNotUpdateableFields(metadataEntry);
-                this.client.SoapClient.update(
-                    this.definition.type.charAt(0).toUpperCase() + this.definition.type.slice(1),
-                    metadataEntry,
-                    null,
-                    (error, response) => (error ? reject(error) : resolve(response))
-                );
-            });
-            Util.logger.info(
-                `- updated ${this.definition.type}: ${metadataEntry[this.definition.keyField]}`
+            this.removeNotUpdateableFields(metadataEntry);
+            await Util.retryOnError(
+                `Retrying to update ${this.definition.type}: ${
+                    metadataEntry[this.definition.nameField]
+                }`,
+                async () =>
+                    (response = await new Promise((resolve, reject) => {
+                        this.client.SoapClient.update(
+                            overrideType ||
+                                this.definition.type.charAt(0).toUpperCase() +
+                                    this.definition.type.slice(1),
+                            metadataEntry,
+                            null,
+                            (error, response) => (error ? reject(error) : resolve(response))
+                        );
+                    }))
             );
-            return res;
+            if (!handleOutside) {
+                Util.logger.info(
+                    `- updated ${this.definition.type}: ${metadataEntry[this.definition.keyField]}`
+                );
+            }
+            return response;
         } catch (ex) {
-            Util.logger.error(
-                `- error updating ${this.definition.type}: ${
-                    metadataEntry[this.definition.keyField]
-                } (${ex.message})`
-            );
+            if (!handleOutside) {
+                let errorMsg;
+                if (ex.results && ex.results.length) {
+                    errorMsg = `${ex.results[0].StatusMessage} (Code ${ex.results[0].ErrorCode})`;
+                } else {
+                    errorMsg = ex.message;
+                }
+                Util.logger.error(
+                    `- error updating ${this.definition.type} '${
+                        metadataEntry[this.definition.keyField]
+                    }': ${errorMsg}`
+                );
+            } else {
+                throw ex;
+            }
 
-            return null;
+            return {};
```

Comment:
sure?

Reply:
same as above. outside scripts actually failed badly with the null response - which we simply didn't see in the past when APIs worked more or less like a charm :D